### ISSUE TITLE
Class Member List does not show friend operator parameter list

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10655]
Class Member List does not show friend operator parameter list.  doxygen-1.9.1 does not generate the warning but doxygen-1.9.2 and later does, including trunk 1.11.0 (4c99c07d2eeaf5284a040ef7995e0998106f5063*).

```
/**
 * Test docs
 */
class Test
{
    /**
     * operator== docs
     */
    friend bool operator==(Test&, Test&);
};
```

Doxygen-1.9.1:
![image](https://github.com/doxygen/doxygen/assets/32584126/9c68bbec-d679-4cd0-982a-733df983443f)

Doxygen-1.9.2 and trunk (4c99c07d2eeaf5284a040ef7995e0998106f5063*):
![image](https://github.com/doxygen/doxygen/assets/32584126/5c5d8c71-2e3f-4104-bc4d-1425d571ed5e)

Testcase:
[testcase.zip](https://github.com/doxygen/doxygen/files/14285273/testcase.zip)

**Comment by albert-github** [comment]
I can see the difference in the output (html/class_test-members.html)  but I don't get a warning with the given example.
- what is the warning you get?


A bisection gave:
```
592aaa4f17d73ec8c475df0f44efaea8cc4d575c is the first bad commit
commit 592aaa4f17d73ec8c475df0f44efaea8cc4d575c

Date:   Sun Apr 11 21:22:59 2021 +0200

    Refactoring: remove implicit conversion from QCString to const char *
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10657

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
